src/foundry/foundry.js/avSettings.d.ts

```diff
@@ -22,9 +22,7 @@ declare class AVSettings {
   };
 
   static VOICE_MODES: {
-    ALWAYS: 'always';
-    ACTIVITY: 'activity';
-    PTT: 'ptt';
+    [x: string]: string;
```

Comment:
This would make it so that `AVSettings.VOICE_MODES['whatever-property-name-you-want']` is of type `string`. That's incorrect as we don't know which properties exist, so it should be `string | undefined`. You would achieve that by
```ts
static VOICE_MODES: Partial<Record<string, string>>;
```
However, by doing that, we lose the concrete type information of the properties that are provided by foundry. foundry seems to operate under the assumption that these exist. In particular, there is even getters in `AVClient` that use these values: `isVoicePTT`, `isVoiceAlways`, `isVoiceActivated`. From my perspective, it doesn't really make sense to remove any of the values `VOICE_MODES`. Adding other voide modes _might_ make sense, so I would be ok with doing something like
```ts
  static VOICE_MODES: {
    ALWAYS: 'always';
    ACTIVITY: 'activity';
    PTT: 'ptt';
  } & Partial<Record<string, string>>;
```
Alternatively we could create an interface `VoiceModes` that you can extend with declaration merging. That might be better actually.

Reply:
Activity voice mode is incompatible with some AV clients. For example, Jitsi and a new modules I am creating both have built in noise cancelation which activity mode steps on and causes cutouts. It also doesn't play nicely with SFU RTC systems in general as you don't have a separate voice stream that can be actively monitored by a level checker. In these cases, ACTIVITY is removed so it does not appear as an option that can be selected by users.

Reply:
OK, makes sense. I think going with `Partial<Record<string, string>>` should be fine then. Do you only remove modes? or do you also add some / access them?

What's your opinion on this @UFOMelkor @kmoschcau?

Reply:
If we end up making the AV classes configurable (as far as Foundry allows), wouldn't it be better to just override the static variable then? Of course the superclass has to allow for it, but that would be the best approach.